src/ephemeris/_split_data_manager_genomes.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+"""Helper script for IDC - not yet meant for public consumption.
+
+This script splits genomes.yml into tasks that are meant to be sent to
+run_data_managers.py - while excluding data managers executions specified
+by genomes.yml that have already been executed and appear in the target
+installed data table configuration.
+"""
+import logging
+import os
+import re
+from copy import deepcopy
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+)
+
+import yaml
+from galaxy.util import safe_makedirs
+from pydantic import BaseModel
+
+from . import get_galaxy_connection
+from .common_parser import (
+    get_common_args,
+)
+from ._data_managers_to_tools import (
+    DataManager,
+    read_data_managers_configuration,
+)
+from .ephemeris_log import (
+    disable_external_library_logging,
+    setup_global_logger,
+)
+
+IsBuildComplete = Callable[[str, str], bool]
+TASK_FILE_NAME = "run_data_managers.yaml"
+
+
+class SplitOptions:
+    merged_genomes_path: str
+    split_genomes_path: str
+    data_managers_path: str
+    is_build_complete: IsBuildComplete
+
+
+def tool_id_for(indexer: str, data_managers: Dict[str, DataManager]) -> str:
+    data_manager = data_managers[indexer]
+    assert data_manager, f"Could not find a target data manager for indexer name {indexer}"
+    return data_manager.tool_id
+
+
+class RunDataManager(BaseModel):
+    id: str
+    items: Optional[List[Any]] = None
+    params: Optional[List[Any]] = None
+    data_table_reload: Optional[List[str]] = None
```

Comment:
This shouldn't be needed. It just works on normal instances, and if it is data on cvmfs you first have to import the bundles this depends on, so the reload also doesn't do anything there.

Reply:
I'm trying to model the inputs these tools consume... if we don't want to use the field that is fine I can drop the dependencies setup with this at line at line 122 but I think keeping the model correct is the right thing to do. Unless we update the samples and run_data_managers.py and such.

Reply:
I would like to migrate all the models out and reuse them in ephemeris and I would like to setup models for genomes.yml and tools.yaml and such... there is some broader context here. I hope you're correct about the reloads not being needed though - ideally those would just happen (see my comment in IDC gitter)